rest/src/main/scala/io/udash/rest/DefaultRestClient.scala

```diff
@@ -0,0 +1,70 @@
+package io.udash
+package rest
+
+import com.avsystem.commons._
+import com.avsystem.commons.annotation.explicitGenerics
+import com.avsystem.commons.meta.Mapping
+import com.softwaremill.sttp.Uri.QueryFragment.KeyValue
+import com.softwaremill.sttp.Uri.QueryFragmentEncoding
+import com.softwaremill.sttp._
+import io.udash.rest.DefaultSttpBackend.backend
+import io.udash.rest.raw._
+
+object DefaultRestClient {
+  @explicitGenerics def apply[RestApi: RawRest.AsRealRpc : RestMetadata](baseUri: String): RestApi =
+    RawRest.fromHandleRequest[RestApi](asHandleRequest(baseUri))
+
+  /**
+    * Creates a [[io.udash.rest.raw.RawRest.HandleRequest HandleRequest]] function which sends REST requests to
+    * a specified base URI using default HTTP client implementation (sttp).
+    */
+  def asHandleRequest(baseUri: String): RawRest.HandleRequest =
+    asHandleRequest(uri"$baseUri")
+
+  private def toSttpRequest(baseUri: Uri, request: RestRequest): Request[String, Nothing] = {
+    val uri = baseUri |>
```

Comment:
Can't we use sttp uri interpolation to make this less of an exercise to read?

Reply:
Please suggest a solution.

Reply:
I'd argue this is more readable:
```scala
    val pathParams = request.parameters.path.map(_.value)
    val queryParams = request.parameters.query.mapValues(_.value)
    val uri = uri"$baseUri/$pathParams?$queryParams"
```
Passes the tests, no modifications needed.

Reply:
Does it work if `baseUri` already has path suffix and query params?

Reply:
I believe they treat the first interpolated argument specially, but you'd have to add a test case.

Reply:
```scala
val baseUri = "http://localhost:8080/fag/lol/dafuq?foo=bar&kek=lel"
val queryParams = Map("a" -> "b", "c" -> "d")
val pathSegs = List("fuuu", "baar")
println(uri"$baseUri/$pathSegs?$queryParams")
```
gives:
```
http://localhost:8080/fag/lol/dafuq?foo=bar&kek=lel/List(fuuu,+baar)?Map(a+-%3E+b,+c+-%3E+d)
```
😕

Reply:
It should be possible to specify URI "prefix", i.e. `"scheme://host:port/basePath"`. I don't care about "base query parameters" and indeed things work well when they are omitted. 

However, in this case we should do some validation so that no abomination like the one above gets generated accidentally. I'm also not sure how the interpolation handles URL encoding.
I must say that with the raw version that I wrote I feel the most confident because it has all these details in it, despite being syntactically inferior.

If you have the patience to rewrite and test this, feel free!